### ISSUE TITLE
BUG - Player card is getting errors after changes in fetching 

### ISSUE THREAD

**Issue by petrvecera** [cohstats/coh3-stats#167]
After we deployed changes to how the player card is fetched, there is increased number of 500 errors. 
<img width="715" alt="image" src="https://user-images.githubusercontent.com/8086995/230055987-84020165-3247-4106-9f4d-2855a1b63346.png">
Changes: https://github.com/cohstats/coh3-stats/pull/152/files

All of them are coming from the the player card:
<img width="1430" alt="image" src="https://user-images.githubusercontent.com/8086995/230056258-704818f4-1404-4593-8bd0-58b0b9aa35bb.png">

This is for the user who visit the player card directly, so we must have something wrong here:
https://github.com/cohstats/coh3-stats/blob/master/pages/players/%5BplayerID%5D.tsx#L188

**Comment by petrvecera** [comment]
The errors are still there, they are coming from server trying to render that page SSR and missing the data. 
<img width="1351" alt="image" src="https://user-images.githubusercontent.com/8086995/231407407-e9316f13-326d-49a0-abc9-cd5f9bd3587f.png">

I am not sure that we can do partial hydratation like this: https://github.com/cohstats/coh3-stats/blob/master/pages/players/%5BplayerID%5D.tsx#L210

It looks like those features are available only with the new /app folder with Next13, where you can have server components. 

Even when we deployed this split, I don't think I see difference in less executions of getting player stats.
<img width="895" alt="image" src="https://user-images.githubusercontent.com/8086995/231406556-c9396298-4b1e-4aa2-89bf-032645f2ab04.png">

I think we need to do a different approach for this whole page. I am gonna try to think about something. 

CC:  @komnb14

**Comment by petrvecera** [comment]
I removed the custom code in our components and added cache layer in front of the cloud functions:
- The errors are completely gone 
- The cache hit ratio is about 30%, cache is for 30 seconds , it adds about ~50 ms extra to the request cos I cache it via another CDN 😆 
- The function executions are little bit lower but I need to monitor that in a long term 

About ~14 days ago I deployed protection against bots on dev domain. It turned out that huge amount of player card executions were caused by fck bing bot who managed to index every single player on the platform 😮‍💨  which is not wrong but he managed to do it on the dev. domain 🤦  even after I added no index nofollow headers.

<img width="1480" alt="image" src="https://user-images.githubusercontent.com/8086995/233023486-127a1293-050f-4ad9-b080-e8e667e76198.png">

<img width="758" alt="image" src="https://user-images.githubusercontent.com/8086995/233022557-ff0d2c3f-a0b9-41e0-984c-6eae1d4a9a56.png">

<img width="671" alt="image" src="https://user-images.githubusercontent.com/8086995/233023937-c43047b5-ed46-4063-9d4e-877072be57de.png">

CC: @komnb14 thanks for helping tackle this 🙏
